common/models/user.js

```diff
@@ -584,7 +584,11 @@ module.exports = function(User) {
   User.hashPassword = function(plain) {
     this.validatePassword(plain);
     var salt = bcrypt.genSaltSync(this.settings.saltWorkFactor || SALT_WORK_FACTOR);
-    return bcrypt.hashSync(plain, salt);
+    if (salt.length < 73) {
+      return bcrypt.hashSync(plain, salt);
+    } else {
+      throw new Error('Validation Error');
```

Comment:
Notice that `validatePassword` provides a helpful error message and HTTP status code 422. We should do the same here.